plugins/mentions/plugin.js

```diff
@@ -142,7 +142,8 @@
 	};
 
 	function createPattern( marker, minChars ) {
-		var pattern = '\\' + marker + '\\w';
+		// Match also diacritic characters (#2491).
+		var pattern = '\\' + marker + '[_A-z0-9À-ž]';
```

Comment:
Let's be more precise with the regexp, your regexp catches also chars like `[`, `^`, backtick etc. `a-zA-Z` is the way to go here.

Reply:
Please add a unit test ensuring that phrases with [^ are not matched.